### ISSUE TITLE
xmlhttp error on Firefox

### ISSUE THREAD

**Issue by WireCrazii** [Kayomani/PlexExternalPlayer#22]
Windows 10, Firefox 58(latest)

```
Plex External Player
Error getting metadata from http://192.168.10.220:32400/library/metadata/18486?checkFiles=1&includeExtras=1Error: ReferenceError: GM_xmlhttpRequest is not defined
```

**Comment by enchained** [comment]
Hm, that's weird. Works for me on 58.0.2. Could you please doublecheck your current script? In Tampermonkey go to Dashboard, click Plex External Player and look for the  `// @grant       GM_xmlhttpRequest` line inside the `==UserScript==` area.
Should be similar for Greasemonkey.

My guess is it's either not granted for some reason, or that's just new Firefox vs old extensions drama. Please update Tamper/Greasemonkey and Firefox if possible, and try again.

**Comment by WireCrazii** [comment]
@enchained Thank you for the quick reply. All of my addons plus Firefox is up to date. And my script looks like this:

```
// ==UserScript==
// @name         Plex External Player
// @namespace    https://github.com/Kayomani/PlexExternalPlayer
// @version      1.15
// @description  Play plex videos in an external player
// @author       Kayomani
// @include     /^https?://.*:32400/web.*
// @include     http://*:32400/web/index.html*
// @require     http://code.jquery.com/jquery-3.2.1.min.js
// @connect     *
// @require     https://cdnjs.cloudflare.com/ajax/libs/toastr.js/latest/js/toastr.min.js
// @grant       GM_xmlhttpRequest
// ==/UserScript==
```

**Comment by Kayomani** [comment]
What version of tampermonkey do you have?

**Comment by WireCrazii** [comment]
@Kayomani I'm sorry it's Greasemonkey 4.2 not Tampermonkey. And Firefox version is 58.0.2(64bit).

**Comment by Kayomani** [comment]
Greasemonkey 4 isn't supported yet, use tampermoney.

**Comment by WireCrazii** [comment]
@Kayomani I see, didn't saw the README was updated. Installed Tampermonkey 4.5.5660 and got another errors. Should I make another issue?

The error messages are:
```
Plex External Player
Playing C:\movies\Video Folder\Video.mkv
```
```
Plex External Player
Could not find authentication info
```
```
Plex External Player
Failed to connect to agent, is it running or firewalled?
```

**Comment by enchained** [comment]
@WireCrazii what about [Plex External] logs in the JS console when this happens?

I actually did have the auth error before, but it went away by itself I think, so you can try generic things like clear cookies and cache, restart browser, log out of plex and login again.

**Comment by WireCrazii** [comment]
@enchained Tried all of them still getting same error messages. I've also tried on chromium(64.0.3282.119) without any extensions and same error messages. Here is the log message from Firefox:

```
[Plex External] Calling http://192.168.10.220:32400/library/metadata/17687?checkFiles=1&includeExtras=1 clientId 5ghkmgzp8ifwmjoce2igtoz2Token xiGz4m2HsBjXtSMfZ4qB
userscript.html:69:5
[Plex External] Called sucessfully to http://192.168.10.220:32400/library/metadata/17687?checkFiles=1&includeExtras=1
userscript.html:69:5
[Plex External] Playing C:\movies\Video Folder\Video.avi
userscript.html:69:5
[Plex External] Calling http://localhost:7251/?protocol=2&item=C%3A%5Cmovies%5CVideo%20Folder%5CVideo.avi clientId 5ghkmgzp8ifwmjoce2igtoz2Token xiGz4m2HsBjXtSMfZ4qB
userscript.html:69:5
[Plex External] Could not find authentication info
userscript.html:69:5
[Plex External] Failed to connect to agent, is it running or firewalled?
```

**Comment by enchained** [comment]
@WireCrazii 
Looks like there's something with the agent indeed (and from the code it looked like the auth error happens only when there's wrong plex user/server info). 
Is the agent running in the tray? Did you try to restart it? Add it to the firewall exceptions?

When the agent works, if you enter that http://localhost:7251/?protocol=2&item=C%3A%5Cmovies%5CVideo%20Folder%5CVideo.avi url into a browser, it'll normally say Recieved.

**Comment by WireCrazii** [comment]
@enchained Yes, the agent is running in the tray. Restarted few times, firewall exceptions added.

Yes the normal download link works without a problem.

I've also tried on different client(Windows 7, Firefox) computer still same messages.

**Comment by enchained** [comment]
@WireCrazii Do you have localhost whitelisted in the Tampermonkey script settings? Go to Dashboard - Plex External Player - Settings - XHR Security. **User domain whitelist** field should contain localhost entry.

**Comment by WireCrazii** [comment]
@enchained Looks localhost already in the whitelist.

**Comment by M-RiC-French** [comment]
Bonjour, 
J'ai exactement le même problème. Avez-vous trouvé une solution ?
En tout cas, même si ça marche pas encore pour moi, merci de vous être penché sur ce besoin d'un lecteur externe pour PLex !!!
J'espère que ça va fonctionner ... même si j'ai effectué toutes les démarches décrites ici et que ça ne fonctionne pas actuellement ...

**Comment by M-RiC-French** [comment]
Hello,
I have exactly the same problem with Plex 1.14.0.5465 + Firefox Quantum 63.0.3 (64 bits) + TamperMonkey 4.8.5847 OR with Chrome 70.0.3538.102 + TamperMonkey 4.7.44. Did you find a solution ?
In any case, even if it does not work for me yet, thank you for considering this need for an external player for PLex!
I hope it will work ... although I have done all the steps described here and it does not work now ...

**Comment by M-RiC-French** [comment]
Works fine, I reinstalled PlexExternalPlayerAgentInstall.v1.12.0.exe from https://github.com/Kayomani/PlexExternalPlayer/releases
